### ISSUE TITLE
crates.io style registry for tremor applications (investigation)

### ISSUE THREAD

**Issue by Licenser** [tremor-rs/tremor-runtime#2700]
**Describe the problem you are trying to solve**

With applications coming in it would be nice to have a registry for applications. crates.io does an excellent job on this and will be a good starting point to learn from

**Describe the solution you'd like**

**Notes**

If this is an implementation of an [RFC](https://github.com/tremor-rs/tremor-rfcs) provide a URL
to the RFC this enhancement implements.

If this is a major enhancement or contribution an RFC may be required. It is ok to submit an enhancement
first and our core team will assist with major contributions. In general, major contributions should be
discussed with the community before submission.

**Comment by devskillz47** [comment]
would this imply that the packages would be published to an online registry hosted somewhere? also, would we have to create a package manager like cargo, pip, npm, etc to pull in the specified dependencies? :-?

**Comment by Licenser** [comment]
Yes we'd need to host the files somewhere - I suspect the CNCF will sponsor this. But I think we can skip on the dependencies for the time being, applications have no concept of that (yet?)
